src/shell/mpir/rangelist.c

```diff
@@ -254,6 +258,44 @@ json_t *range_to_json (struct range *r, int64_t base)
     return o;
 }
 
+static json_int_t range_json_val (json_t *array, size_t index)
+{
+    return json_integer_value (json_array_get (array, index));
+}
+
+static bool range_json_equal (json_t *a, json_t *b)
+{
+    if (range_json_val (a, 0) == range_json_val (b, 0)
+        && range_json_val (a, 1) == range_json_val (b, 1))
+        return true;
+    return false;
+}
+
+static int increment_range_repeat (json_t *range)
+{
+    if (json_array_size (range) == 3) {
+        json_int_t repeat = range_json_val (range, 2);
+        return json_array_set_new (range, 2, json_integer (repeat + 1));
+    }
+    return json_array_append_new (range, json_integer (1));
+}
```

Comment:
If either of the `_new()` functions fails, the `json_integer()` value is leaked.